tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -355,6 +358,8 @@ static PyObject * THPVariable_sparse_coo_tensor(PyObject* self, PyObject* args,
   END_HANDLE_TH_ERRORS
 }
 
+// bound here bc it is more performant to construct the tensor while recursing through
```

Comment:
This isn't the main reason, right? `data` is pretty dynamic (it can be any type of array like object -- list, tuple, np array, scalar) and we are unable to represent that in native_functions

Reply:
^^ yes agreed